packages-native/listview-swipe/src/ListViewSwipe.tsx

```diff
@@ -1,101 +1,130 @@
 import Swipeable from "react-native-gesture-handler/Swipeable";
 import { defaultListViewSwipeStyle, ListViewSwipeStyle, PanelStyle } from "./ui/styles";
+import { ActionValue } from "mendix";
 
-export class ListViewSwipe extends Component<ListViewSwipeProps<ListViewSwipeStyle>> {
-    private row: any;
-    private readonly styles = flattenStyles(defaultListViewSwipeStyle, this.props.style);
-    private readonly onSwipeLeftHandler = this.onSwipeLeft.bind(this);
-    private readonly onSwipeRightHandler = this.onSwipeRight.bind(this);
+export const ListViewSwipe = (props: ListViewSwipeProps<ListViewSwipeStyle>): ReactElement => {
+    const row = useRef<Swipeable>(null);
+    const styles = flattenStyles(defaultListViewSwipeStyle, props.style);
+    const [animation, setAnimation] = useState(new Animated.Value(0));
+    const [animate, setAnimate] = useState(false);
 
-    private readonly isLeftSideAction = this.props.leftRenderMode === "action";
-    private readonly isRightSideAction = this.props.rightRenderMode === "action";
-    private readonly isLeftDisabled = this.props.leftRenderMode === "disabled";
-    private readonly isRightDisabled = this.props.rightRenderMode === "disabled";
+    const isLeftSideAction = props.leftRenderMode !== "buttons";
+    const isRightSideAction = props.rightRenderMode !== "buttons";
+    const isLeftDisabled = props.leftRenderMode === "disabled";
+    const isRightDisabled = props.rightRenderMode === "disabled";
+    const isLeftToggle = props.leftRenderMode === "toggle";
+    const isRightToggle = props.rightRenderMode === "toggle";
 
-    render(): ReactNode {
-        return (
-            <Swipeable
-                ref={this.updateRef}
-                friction={1}
-                leftThreshold={this.isLeftSideAction ? 100 : 40}
-                rightThreshold={this.isRightSideAction ? 100 : 40}
-                renderLeftActions={this.renderLeftActions}
-                renderRightActions={this.renderRightActions}
-                onSwipeableLeftOpen={this.onSwipeLeftHandler}
-                onSwipeableRightOpen={this.onSwipeRightHandler}
-                overshootLeft={false}
-                overshootRight={false}
-                useNativeAnimations
-            >
-                <View style={this.styles.container}>{this.props.content}</View>
-            </Swipeable>
-        );
-    }
+    const setAnimationDefinitions = useCallback((event: any) => {
+        setAnimation(new Animated.Value(event.nativeEvent.layout.height));
+    }, []);
 
-    renderLeftActions = (): ReactNode => {
-        if (this.isLeftDisabled) {
+    const renderLeftActions = (): ReactNode => {
+        if (isLeftDisabled) {
             return undefined;
-        } else if (this.isLeftSideAction) {
-            return this.renderAction(this.styles.leftAction, this.props.left);
+        } else if (isLeftSideAction) {
+            return renderAction(styles.leftAction, props.left, isLeftToggle);
         } else {
```

Comment:
Else statement not necessary, because of returns in the other conditonal statements.

Reply:
Done